code/src/components/CategoryMap.js

```diff
@@ -0,0 +1,184 @@
+  background-image: url(${img});
+  background-repeat: no-repeat;
+  background-size: 15px;
+  background-position: 95% center; 
+  margin-left: 8px;
+  @media (min-width: 660px) {
+     background-size: 20px;
+   }
+`
+export const CategoryMap = () => {
+    const dispatch = useDispatch()
+    const folders = useSelector((store) => store.todos.list.categories)
+
+    const openFolder = (itemIndex, item) => {
+        dispatch(
+            todos.actions.revealItem({ itemIndex: itemIndex, reveal: !item.reveal })
+        )
+    }
+
+    const sortFolder = (value, index) => {
+        dispatch(
+            todos.actions.sortFolder({ itemIndex: index, sortOrder: value })
+        )
+
+    }
+
+    const clearFolder = (index) => {
+        dispatch(
+            todos.actions.clearFolder({ itemIndex: index })
+        )
+    }
+
+    const removeCategory = (index) => {
+        dispatch(
+            todos.actions.removeCategory({ itemIndex: index })
+        )
+    }
+
+    return (
+        folders.map((folder, index) => {
+            const sortByCreated = folder.items.filter(x => x.created)
+            sortByCreated.sort((a, b) => (a.done === b.done) ? ((a.created > b.created) ? 1 : -1) : a.done ? 1 : -1)
+            const unFinished = folder.items.filter(x => !x.done)
+            const dueDateArray = folder.items.filter((x => x.dueDate))
+            dueDateArray.sort((a, b) => (a.done === b.done) ? ((a.dueDate > b.dueDate) ? 1 : -1) : !a.done ? 1 : -1)
+            const noDateArray = folder.items.filter((x => !x.dueDate))
+            const sortByDueDate = dueDateArray.concat(noDateArray)
+            sortByDueDate.sort((a, b) => (a.done === b.done) ? 0 : a.done ? 1 : -1)
+            return (
+                <LightSpeed left>
+                    <section key={index}>
+                        <FolderContainer backgroundColor={folder.reveal ? "papayawhip" : "#FCFCFC"}>
+                            <LeftSideContainer>
+                                {unFinished.length !== 0 &&
+                                    <Bounce >
+                                        <Dot></Dot>
+                                    </Bounce>}
+                                <FolderButton onClick={() => openFolder(index, folder)}
+                                    fontSize={folder.name.length > 16 ? "13px" : "16px"}>
+                                    <Image src={require('../Assets/FolderIcon.png')} />
+                                    <ButtonText>
+                                        {folder.name} ({folder.items.length} {folder.items.length === 1 ? "item" : "items"})
+                                </ButtonText>
+                                </FolderButton>
+                                <DropdownContainer>
+                                    <Dropdown onChange={(event) => event.target.value === "dueDate" ? sortFolder(sortByDueDate, index) : event.target.value === "dateCreated" ? sortFolder(sortByCreated, index) : ""}>
```

Comment:
I would try to extract this a bit to make it more readable.
This is totally fine though, but it can be a bit tricky to read when nesting ternaries..
```
const handleSortFolder = (event) => {
 sortFolder(event.target.value, index)
}

<Dropdown onChange={(event) => handleSortFolder(event)}>
```